generators/generate

```diff
@@ -28,12 +28,16 @@ def download_canonical_data(exercise):
 
 def flatten_cases(data):
     cases_by_id = {}
-    for i in data["cases"]:
-        if "cases" in i:
-            for j in i["cases"]:
-                add_case(cases_by_id, j)
+
+    def traverse(node):
+        nonlocal cases_by_id
+        if "cases" in node:
+            for child in node["cases"]:
+                traverse(child)
         else:
-            add_case(cases_by_id, i)
+            add_case(cases_by_id, node)
+
+    traverse(data)
```

Comment:
This set of changes seems to be unrelated to the purpose of the pull request.

Reply:
The generator previously did not allow for deeply nested tests.

If I omit this change, we have

```
Traceback (most recent call last):
  File "generators/generate", line 111, in <module>
    main()
  File "generators/generate", line 96, in main
    cases = flatten_cases(data)
            ^^^^^^^^^^^^^^^^^^^
  File "generators/generate", line 34, in flatten_cases
    add_case(cases_by_id, j)
  File "generators/generate", line 44, in add_case
    cases_by_id[case["uuid"]] = case
                ~~~~^^^^^^^^
KeyError: 'uuid'

```


If I just made a one-line generator change

```
 def add_case(cases_by_id, case):
     if "reimplements" in case:
         cases_by_id[case["reimplements"]] = case
-     else:
+     elif "uuid" in case:
         cases_by_id[case["uuid"]] = case
```

then generation reports no errors, but omits two test cases:

```
-    RUN_TEST(test_second_students_garden);
-    RUN_TEST(test_third_students_garden);
```

Reply:
I did not see a separate commit for this explaining that.

Do we need maybe reform the patch and force push on this feature branch to update and separate in case one thing needs to be reverted  but not the other things that may be good?

Reply:
Extracted https://github.com/exercism/x86-64-assembly/pull/249    that can be merged first.

Reply:
Because of the #249 are we going to modify this patch to exclude it?  It still should not conflict, if the other is done first, but as it is it seems "off" just a bit.

I do not have enough rights on this repository to fix it up, otherwise I would do that, instead of tossing the bone back your way.

Reply:
Now modified.